p2p/chain.py

```diff
@@ -112,13 +112,19 @@ def register_peer(self, peer: BasePeer) -> None:
 
     async def handle_msg(self, peer: ETHPeer, cmd: protocol.Command,
                          msg: protocol._DecodedMsgType) -> None:
+        known_exceptions = (
+            BrokenPipeError,
+            ConnectionResetError,
+        )
         try:
             await self._handle_msg(peer, cmd, msg)
         except OperationCancelled:
             # Silently swallow OperationCancelled exceptions because we run unsupervised (i.e.
             # with ensure_future()). Our caller will also get an OperationCancelled anyway, and
             # there it will be handled.
             pass
+        except known_exceptions as exc:
+            self.logger.warn("Halted while handling message from %s with %r", peer, exc)
```

Comment:
I'd really prefer if those were caught closer to the code that originated them, as these should definitely not be bubbling up from `Peer` methods, so I suspect they're coming from trinity's proxy services when they're shut down and this would not be needed if we were able to shut everything down cleanly ([which is currently not the case](https://github.com/ethereum/py-evm/issues/665))

Reply:
Fair enough. Unfortunately one crash is causing a snowball of all the other cleanups to fail. 

I'll log out the full exception, which will have almost the same effect as not catching it (but will allow the cleanups to complete).